### ISSUE TITLE
Select first Task to complete

### ISSUE THREAD

**Issue by cquiroz** [scalaz/scalaz#1277]
I have a use case where I have multiple tasks that I want to execute in parallel, I only care about the first result to complete, and I can let the other tasks to continue in the background. I also don't care about failures as long as one of my tasks completes. I should only get an error if all of my tasks error out

None of the current methods (`gatherUnordered`, `chooseAny`) works for me, mostly due to the way errors are handled

I came up with this solution but I'm not sure if it is fully correct. Does it look correct to you? I'd think this is generally useful

```scala
  /**
   * From a list of task create a new Task that will return the first available result
   * or a failure if all the tasks fail
   */
  def selectFirstToComplete[A](tasks: NonEmptyList[Task[A]])(implicit pool: ExecutorService): Task[A] = {
    tasks match {
      case NonEmptyList(t, INil()) => t
      case _                       =>
        // Create a new task with a Future.Async and call the callback when appropriate
        new Task(Future.Async[Throwable \/ A](callback => {
          val interrupt = new AtomicBoolean(false)
          val failures = new AtomicInteger(tasks.size)
          val handle: (Throwable \/ A) => Unit = {
            case s @ \/-(_) if interrupt.compareAndSet(false, true) =>
              callback(s).run
            case e @ -\/(_) if failures.decrementAndGet() == 0      => // If all fail return the failure
              callback(e).run
            case _                                                  => // Other failures are ignored
          }
          // Call each task in its own thread on the pool
          tasks.map(Task.fork(_)(pool).unsafePerformAsyncInterruptibly(handle, interrupt))
        }))
    }
  }
```
